django/db/models/expressions.py

```diff
@@ -485,12 +485,23 @@ def __init__(self, value, output_field=None):
         self.value = value
 
     def as_sql(self, compiler, connection):
-        if self.value is None:
+        val = self.value
+        if self._output_field_or_none is not None:
+            if self.for_save:
```

Comment:
That's what I wanted to do originally. However not every ``Value`` in an update is for saving. For example in ``When`` only the result expression needs to get ``for_save`` from the parent expression, the condition expression always gets ``for_save=False``.